src/CounterInterceptor/SimpleClientCallsInterceptor.cs

```diff
@@ -0,0 +1,97 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using Temporalio.Client;
+using Temporalio.Client.Interceptors;
+
+public record ClientCounts
+{
+    public uint Executions { get; internal set; }
+
+    public uint Signals { get; internal set; }
+
+    public uint Queries { get; internal set; }
+
+    public override string ToString() =>
+        $"\n\tTotal Number of Workflow Exec: {Executions}\n\t" +
+        $"Total Number of Signals: {Signals}\n\t" +
+        $"Total Number of Queries: {Queries}";
+}
+
+public class SimpleClientCallsInterceptor : IClientInterceptor
+{
+    private const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    private const string NumberOfSignals = "numOfSignals";
+    private const string NumberOfQueries = "numOfQueries";
+    private static Dictionary<string, ClientCounts> clientDictionary = new();
+
+    public ClientOutboundInterceptor InterceptClient(ClientOutboundInterceptor nextInterceptor) =>
+        new ClientOutbound(this, nextInterceptor);
+
+    public string Info() =>
+        string.Join(
+            "\n",
+            clientDictionary.Select(kvp => $"** Workflow ID: {kvp.Key} {kvp.Value}"));
+
+    public uint NumOfWorkflowExecutions(string workflowId) =>
+        clientDictionary[workflowId].Executions;
+
+    public uint NumOfSignals(string workflowId) =>
+        clientDictionary[workflowId].Signals;
+
+    public uint NumOfQueries(string workflowId) =>
+        clientDictionary[workflowId].Queries;
+
+    private void Add(string workflowId, string type)
```

Comment:
This is not thread safe. What if multiple threads for the same workflow ID come in at the same time, there is a case where one dictionary value may get overwritten with another. Consider a `ConcurrentDictionary`.